### DISCUSSION TITLE
Chromeless window with transparency

### DISCUSSION THREAD

**Discussion by ZodmanPerth** [dotMorten/WinUIEx#134] [Q&A] [answered]
I'm looking to create a window with no chrome and rounded corners.  I've been able to add transparency to the window but I can't completely get rid of the chrome.

i.e. I'm trying to get this type of look for my window (taken from the release notes):
![image](https://github.com/dotMorten/WinUIEx/assets/546368/ea7f856a-e8ca-4608-9fec-25c319be1ae3)

The best I've been able to manage is to set the `WindowStyle` to `Border`,  but there is a faint line around the window.  I've scoured the docs and api reference and I still can't figure out how to get a borderless window with no menubar and transparency.

Can you please give me a clue as to what I'm missing?

**Comment by dotMorten** [comment]
I haven't been able to get rid of the line myself either just yet. If you look closely in the above screenshot you can still see it in the gap between the two parts of the window:  
<img width="62" alt="image" src="https://github.com/dotMorten/WinUIEx/assets/1378165/df24960d-900d-442b-900a-8a56adacfc45">

**Reply by ZodmanPerth** [reply]
Oh, wow.  Damn.
Thanks.

**Comment by dotMorten** [comment, marked as answer]
@riverar likes to say he’s the windows expert so perhaps he will have a few suggestions 😉

**Reply by ZodmanPerth** [reply]
It does appear the rounded corners on the top and bottom panels overlap the thin border.  If that's the case it's probably the resolution for my scenario, where both the background and border colours will have full opacity.  

@riverar are you overlapping the edge of the window with your content?  🙏

**Reply by dotMorten** [reply]
I do know that you might need to mess with the extended styles as well (I recently added support for that in this commit: https://github.com/dotMorten/WinUIEx/commit/21e9ff7abbd62efb9ea235f6b5f1d34f80fc378f when experimenting with it).

Also when working on the custom backdrops, I noticed that there are actually two window handles in play, so another thing I need to look into is whether we need to mess with the second window handle as well.

**Reply by riverar** [reply]
If you can provide a minimal repro, I can take a look/offer some tips.

**Reply by dotMorten** [reply]
Added it to the sample app in the https://github.com/dotMorten/WinUIEx/tree/transparent_test branch.
Just run sample app, go to "Window Design" and click the "open transparent window"

All the interesting bits are right here: https://github.com/dotMorten/WinUIEx/commit/46f94cb33f8169175882fa3ad3c59acd7847ec4e

**Reply by riverar** [reply]
I set `ex:WindowExtensions.WindowStyle="0"` and can't seem to reproduce

![image](https://github.com/dotMorten/WinUIEx/assets/475132/76ca4214-e92e-4fc5-81ae-14b7a05b3666)

**Reply by dotMorten** [reply]
Thanks @riverar . This is the point where I hit myself for realizing this is exactly what I do in the splashscreen too!
https://github.com/dotMorten/WinUIEx/blob/282a04a2b10846ea81d9ba20486be02d8077525a/src/WinUIEx/SplashScreen.cs#L101

**Reply by ZodmanPerth** [reply]
I've been able to produce a transparent window myself.  Here's a crude repro:

```
// Create some visible window content
var grid = new Grid();
grid.CornerRadius = new CornerRadius(12);
grid.Background = new SolidColorBrush(Colors.AliceBlue);
grid.BorderBrush = new SolidColorBrush(Colors.DarkBlue);
grid.BorderThickness = new Thickness(2);

// Create transparent window
var transparentWindow = new Window();
transparentWindow.SetWindowSize(400, 300);
transparentWindow.SystemBackdrop = new TransparentTintBackdrop();
transparentWindow.SetIsShownInSwitchers(false);
transparentWindow.SetWindowStyle(WindowStyle.Tiled);

// Display content in window
transparentWindow.Content = grid;
transparentWindow.Show();
```

In my defence I never thought to try using a `Tiled` window style because the both the name and the intellisense led me away from using it:
```
/// <summary>The window is an overlapped window. An overlapped window has a title bar and a border. Same as the OVERLAPPED style.</summary>
Tiled = 0x00000000,
```

I did hunt around for docs about the Overlapped style but didn't find any.  I just assumed it was for a scenario I wasn't interested in and ignored `Tiled`.

In the interests of discoverability, perhaps updating the summary of `Tiled` to include some text about this style allowing transparent windows, or using another style enum value `Transparent` with a value of `0x00000000`, would help people like me looking for transparent window support.

Thank you both for helping me out with this.

**Reply by dotMorten** [reply]
> In the interests of discoverability, perhaps updating the summary of Tiled to include some text about this style

Doc is verbatim from here: https://learn.microsoft.com/en-us/windows/win32/winmsg/window-styles
